.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,128 @@
+name: 'Start OpenSearch'
+description: 'Downloads latest build of OpenSearch, installs a plugin, executes and script and then starts OpenSearch on localhost:9200'
+
+inputs:
+  platform:
+    description: 'What platform is this action running on, e.g. "windows-latest" ? '
+    required: true
+
+  opensearch-version:
+    description: 'The version of OpenSearch that should be used, e.g "3.0.0"'
+    required: true
+
+  plugin-name:
+    description: 'The the name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+  plugin-start-script:
+    description: 'The file name for the configuration script for the plugin such as install_demo_configurations -- may not be needed for every plugin'
+    required: false
+
+  plugin-zip:
+    description: 'The name of the zip file for the plugin hosted on docker-host i.e. security-plugin.zip '
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    # Download OpenSearch
+    - uses: peternied/download-file@v1
+      if: ${{ inputs.platform == 'windows-latest' }}
+      with:
+        url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
+
+    - uses: peternied/download-file@v1
+      if: ${{ inputs.platform == 'ubuntu-latest' }}
+      with:
+        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+
+    # Extract downloaded zip
+    - name: Extract downloaded zip for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest' }}
+      run: |
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - name: Extract downloaded zip for Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: |
+        tar -xzf opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
+        del opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip
```

Comment:
What do you think of using `*` pattern as above, if you haven't tried already ? (i know PowerShell is a little painful)

Reply:
Yeah I tried, without specifying the whole path I keep getting errors about multiple inputs :/ . I can try again.

Reply:
<img width="1073" alt="Screen Shot 2022-11-09 at 11 59 04 AM" src="https://user-images.githubusercontent.com/65832608/200893123-c206adcb-0ead-4673-bd8c-023c4f20f69f.png">
Foiled again :/